eox_nelp/api_clients/tests/mixins.py

```diff
@@ -151,6 +154,74 @@ def test_failed_get(self, requests_mock):
         ])
 
 
+class TestSOAPClientMixin:
+    """Basic API client tests."""
+
+    @patch("eox_nelp.api_clients.authenticators.requests")
+    def test_successful_post(self, requests_mock):
+        """Test case when a POST request success.
+
+        Expected behavior:
+            - Response is the expected value
+            - POST was called with the given data and right url.
+        """
+        response = Mock()
+        response.ok = True
+        response.text = "<result> xml response string from API </result>"
+        expected_value = BeautifulSoup(response.text, "xml")
+        requests_mock.Session.return_value.post.return_value = response
+        data = {"testing": True, "application": "pearson_vue"}
+
+        with patch.object(self.api_class, "_authenticate") as auth_mock:
+            auth_mock.return_value = requests_mock.Session()
+            api_client = self.api_class()
+
+        response = api_client.make_post("fake/path", data)
```

Comment:
data should be string??
https://github.com/eduNEXT/eox-nelp/blob/e196a43694caf6a47f6e9fb8bf0c4de65b008627/eox_nelp/api_clients/__init__.py#L134-L141

Reply:
Based on the docstring yes, that should be a string, based on the requests [documentation](https://requests.readthedocs.io/en/latest/api/#requests.Session.post) that could be a `Dictionary, list of tuples, bytes, or file-like object` anyway I`ll add a extra verification since this is soap client and I think it's just possible to use strings in that case